evm/chains/base.py

```diff
@@ -142,29 +165,29 @@ def get_canonical_block_by_number(self, block_number):
         validate_uint256(block_number, title="Block Number")
         return self.get_block_by_hash(self.chaindb.lookup_block_hash(block_number))
 
-    def get_block_by_hash(self, block_hash):
+    def get_block_by_hash(self, block_hash: Hash32) -> BaseBlock:
         """
         Returns the requested block as specified by block hash.
         """
         validate_word(block_hash, title="Block Hash")
         block_header = self.get_block_header_by_hash(block_hash)
         return self.get_block_by_header(block_header)
 
-    def get_block_by_header(self, block_header):
+    def get_block_by_header(self, block_header: BlockHeader) -> BaseBlock:
         vm = self.get_vm(block_header)
         return vm.get_block_class().from_header(block_header, self.chaindb)
 
-    @to_tuple
-    def get_ancestors(self, limit):
-        lower_limit = max(self.header.block_number - limit, 0)
-        for n in reversed(range(lower_limit, self.header.block_number)):
-            yield self.get_canonical_block_by_number(n)
+    def get_ancestors(self, limit: int) -> Generator[BaseBlock, None, None]:
```

Comment:
Hard to follow up on mobile but are you sure we t don't want `Iterator[Baseblock] `?